### ISSUE TITLE
Force target to have feature dimension

### ISSUE THREAD

**Issue by kaareendrup** [icecube/graphnet#143]
This issue is a suggestion to add a feature dimension to target tensors and update all loss/task functions etc. to adopt this functionality. 

Right now, the model output has a feature dimension, while the target does not, and the forward call of a typical loss function (here `LogCoshLoss`) includes this syntax:
`assert prediction.dim() == target.dim() + 1`
 `diff = prediction[:,0] - target`

The background for adding the feature dimension to the target, is that this standard syntax in sklearn, which causes a problem when working with more complex transformation functions. When applying a function like `sklearn.preprocessing.QuantileTransformer` to the target, it is necessary to add a dimension before transformation and remove it again afterwards. This is not a major problem (though it does add room for mistakes), but it causes further challenges when doing things like the sanity check in #125 which asserts that the target and inference transformations are inverse.

These points cause me to believe that the developers of sklearn have a good reason for choosing this syntax as standard, and after discussion with @asogaard, I believe that our code will be simpler and more robust if we adopt it.
